src/components/intent.ts

```diff
@@ -464,10 +468,12 @@ export class Intent {
         await this._ensureJoined(roomId);
         await this._ensureHasPowerLevelFor(roomId, type, false);
         return this._joinGuard(roomId, async() => {
-            return {
+            const result = {
                 // eslint-disable-next-line camelcase
                 event_id: await this.botSdkIntent.underlyingClient.sendEvent(roomId, type, content),
-            }
+            };
+            this.opts.onEventSent?.(roomId, type, content, result.event_id);
```

Comment:
Real question, do we count state events?

Reply:
Apparently we don't. Should we? Prior art[tm] did not, from what I can tell.